Breadcrumbs

kosterm14 opened the pull request:
# Breadcrumbs
## Description

Na vraag van de opdrachtgever heb ik op de error pagina een breadcrumbs navigatie toegevgoed zodat de gebruiker weet waar ze vandaan kwamen en hoe ze terug moeten komen op pagina's die wel werken

Issue #128 

![image](https://github.com/fdnd-agency/oba/assets/61830362/24141071-b339-4309-9027-374bcd81a186)

https://mijn-oba.netlify.app/leeslijst/sbhcfhgsjd (werkt pas na de merge)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] User test
- [ ] Accessibility test
- [ ] Performance test
- [ ] Responsive Design test

Ik heb met David en Ivar een user test gedaan over met name de kleur en de duidelijkheid van de navigatie.

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] The code is accessible
- [ ] The code is performant
- [ ] The code is responsive
- [x] I have written meaningful commit messages
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (README and/or Wiki)

kosterm14 committed:
Merge remote-tracking branch 'origin/release-candidate-my-oba' into release-candidate-my-oba-marco

kosterm14 committed:
test

kosterm14 committed:
Error pagina

toevoeging van een error pagina voor als het domein niet bestaat

kosterm14 committed:
error page v1

De basiscode om op verder te werken

kosterm14 committed:
error page v1

De basiscode om op verder te werken

kosterm14 committed:
Update +error.svelte

kosterm14 committed:
Merge branch 'error-page' into release-candidate-my-oba-marco

kosterm14 committed:
Merge pull request #2 from Davidvboxt/release-candidate-my-oba-marco

meest recente code

kosterm14 committed:
Merge pull request #4 from Davidvboxt/release-candidate-my-oba-david

Release candidate my oba david

kosterm14 committed:
Chatbot V1

Eerste versie werkende chatbot. Chatbot reageer op basis van een script, niet met intelligentie

kosterm14 committed:
idk

graphql request geinstalleerd

kosterm14 committed:
breadcrumbs navigatie

Na vraag van de opdrachtgever heb ik op de error pagina een breadcrumbs navigatie toegevgoed zodat de gebruiker weet waar ze vandaan kwamen en hoe ze terug moeten komen op pagina's die wel werken

joostf reviewed (commented):
Wat betekent de commit message 'test'?

joostf commented on src/lib/molecules/footer.svelte:
```diff
@@ -1,57 +1,71 @@
-    </article>
+	<article>
+		<nav>
+			<button on:click={() => navigateTo('/')}>Home</button>
+            <button on:click={() => navigateTo('/leeslijst')}>Leeslijst</button>
+            <button on:click={() => navigateTo('/uitleningen')}>Uitleningen</button>
+            <button on:click={() => navigateTo('/reserveringen')}>Reserveringen</button>
+            <button on:click={() => navigateTo('/betalingen')}>Betalingen</button>
+		</nav>
+	</article>
+	<article>
+		<button on:click={() => navigateTo('/')}>
+            <img src={LogoBlue} alt="oba logo" width="200" />
+        </button>
```
Geen buttons gebruiken voor navigatie naar interne of externe url's. Dit met buttons en clients side javascript oplossen is bad practise; niet toegankelijk, niet robuust (want je maakt geen gebruik van web standaarden) en je maakt het onnodig complex.

Maak een breakdown van het nav element met semantische elementen!

kosterm14 replied:
Als je vanuit de error pagina op de navigatie in de footer drukt werkt de CSS van die pagina niet. Ik heb dit gefixt door dit te doen. Weet jij misschien hoe ik forceer dat de CSS refreshed ofzo als er op een link gedrukt wordt?

ju5tu5 replied:
Hier ben je inmiddels uit gekomen toch?

kosterm14 replied:
ja ik heb de fix al gecommit

joostf commented on src/lib/molecules/footer.svelte:
```diff
@@ -1,57 +1,71 @@
 
-</style>
\ No newline at end of file
+	@media only screen and (min-width: 45rem) {
```
Vervangen door een container query?

kosterm14 replied:
De footer is eigenlijk david's code ik heb hier niet aan gewerkt

joostf commented on src/routes/[...slug]/+error.svelte:
```diff
@@ -0,0 +1,131 @@
+	<nav>
+		<ul>
+			<li>
+				<button class="breadcrumbs {pathParts.length === 0 ? 'current' : ''}" on:click={() => navigateTo('/')}>home</button>
+			</li>
+			{#each pathParts as part, index (index)}
+				<li>
+					<button
+						class="breadcrumbs {isCurrentPage(index) ? 'current' : ''}"
+						on:click={() => navigateTo(`/${pathParts.slice(0, index + 1).join('/')}`)}
+						>{part}</button
+					>
```
Ook voor de breadcrumbs geen buttons gebruiken, maar links. Het kunnen door- en deeplinken naar url's is een van de belangrijkste principes van het web!

joostf replied:
Breadcrumbs in een component zetten en op alle pagina's includen?

kosterm14 replied:
Ik heb hier ook buttons gebruikt omdat dit ook in de error pagina is en hier geen CSS wordt meegegeven om een of andere reden (zie wat ik hier 2 comments boven zei) misschien weet je een oplossing die mijn css wel meegeeft?

Breadcrumbs op alle pagina's is een goed idee ik zal dat zeker doen!

joostf replied:
Ik kijk woensdag even met je mee!

joostf commented on src/routes/[zoekresultaten]/+page.svelte:
Op de zoekresultaten pagina feedback geven over op wat gezocht is en hoeveel resultaten het waren?

<img width="922" alt="feedback-zoeken" src="https://github.com/fdnd-agency/oba/assets/1061632/259a75f4-341e-47ee-bd1c-af0f1cb74a27">

kosterm14 replied:
De zoekresultaten zijn eigenlijk david's code ik heb hier niet aan gewerkt

joostf commented:
@kosterm14 Nette PR, we moeten nog even nadenken over een live  en staging live omgeving. tbc

Deze PR zou in 3 kleinere PR's kunnen worden opgedeeld; breadcrumbs, zoeken en chatbot.

joostf commented on src/routes/chatbot/+page.svelte:
```diff
@@ -0,0 +1,137 @@
+	function sendMessage() {
+		messages.update((currentMessages) => {
+			let updatedMessages = [...currentMessages, { text: message, sender: 'user' }];
+			message = '';
```
Is het nodig om de message binding opnieuw een empty string value te geven? Dat deed je ook op regel 4

kosterm14 replied:
Ja dit is nodig want de code op regel 4 zorgt ervoor dat de message in eerste instantie leeg is en de code die jij referenced zorgt ervoor dat het message veld weer leeg is na het verzenden van een message.

joostf commented on src/routes/chatbot/+page.svelte:
```diff
@@ -0,0 +1,137 @@
+</script>
+
+<main>
+	<div class="chatbox">
+		{#each $messages as msg (msg.text)}
+			<div class={msg.sender}>
+				<p>{msg.text}</p>
+			</div>
+		{/each}
+	</div>
```
Kan je hier semantische HTML van maken?

kosterm14 replied:
Bedoel je hiermee geen divs gebruiken? Ik vervang het wel met iets semantischer

joostf commented on src/routes/chatbot/+page.svelte:
```diff
@@ -0,0 +1,137 @@
+		{/each}
+	</div>
+
+	<input
+		bind:value={message}
+		on:keydown={(e) => {
+			if (e.key === 'Enter') {
+				e.preventDefault();
+				sendMessage();
+			}
+		}}
+		placeholder="Type a message..."
+	/>
+	<button on:click={sendMessage}>Send</button>
```
Hier een formulier omheen zetten?
Dan kan je de chat PE opzetten met SvelteKit Form Actions: 

https://kit.svelte.dev/docs/form-actions

kosterm14 replied:
Ik zal dit zeker doen! Dit was meer een chatbot V1 waarin de focus erop lag dat het werkte. Ik zal in V2 hier meer op letten

joostf commented on src/routes/[zoekresultaten]/+page.svelte:
In je CSS zitten heel veel onnodige px value's. Graag passendere units gebruiken!

Bronnen:
https://techhub.iodigital.com/articles/going-beyond-pixels-and-rems-in-css/relative-length-units-based-on-font
https://www.freecodecamp.org/news/css-units-when-to-use-each-one/

kosterm14 replied:
De zoekresultaten zijn eigenlijk david's code ik heb hier niet aan gewerkt

joostf reviewed (commented):
@kosterm14 
Graag issues van aanmaken en in 3 aparte PR's opnieuw aanbieden: breadcrumbs, zoeken en chatbot.

Als je de issues hebt aangemaakt, merge ik deze PR alvast naar live, dan kan je het per feature verbeteren!

kosterm14 commented:
> Wat betekent de commit message 'test'?

Ik was aan het kijken of de gitignore wel werkte. Ik zal dat duidelijker aangeven volgende keer.

kosterm14 commented:
> @kosterm14 Nette PR, we moeten nog even nadenken over een live en staging live omgeving. tbc
> 
> Deze PR zou in 3 kleinere PR's kunnen worden opgedeeld; breadcrumbs, zoeken en chatbot.

Ik heb ook 2 verschillende pull requests gemaakt. 1 voor de breadcrumbs en 1 voor de chatbot. In die van de chatbot staat ook de error page omdat ik de pull request van de error page in een verkeerde repo had gedaan en die nooit gemerged was. Hierdoor heb ik de code van de error page branch overgezet naar de chatbot branch en hier verder in gewerkt.

Maar breadcrumbs en chatbot hebben beide een aparte PR. De zoekbalk is van david niet van mij dus hier heb ik geen PR voor.

kosterm14 committed:
:sparkles: :zap:  Breadcrumbs component + inline JS removal

Ik heb een component gemaakt van de breadcrumbs en deze op alle pagina's gezet. Hiernaast heb ik de inline javascript in de footer en breadcrumbs verwijderd en deze buttons vervangen door a tags voor meer semantiek in mijn code

ju5tu5 commented:
Prima voor nu, we kijken morgen even verder...
